index.js

```diff
@@ -937,27 +975,32 @@ class ClientServiceBase {
         ],
     );
 
-    argument['nonce'] = uuidv4();
-    const argumentJson = JSON.stringify(argument);
-    const functionArgumentJson = JSON.stringify(functionArgument);
+    const functionIds =
+      typeof functionId === 'string' && functionId.length > 0 ?
+        [functionId] :
+        [];
 
     const builder = new ContractExecutionRequestBuilder(
         new this.protobuf.ContractExecutionRequest(),
         this._createSigner(properties),
-    ).withContractId(contractId)
-        .withContractArgument(argumentJson)
-        .withFunctionArgument(functionArgumentJson)
+    )
+        .withContractId(contractId)
+        .withContractArgument(format(nonce, functionIds, contractArgument))
```

Comment:
We will always serialize the contract argument by v2 format in Node SDK since this change will never be used to send requests to a Scalar DL less than version 3.5.

The implementation of the format function was reviewed in https://github.com/scalar-labs/scalardl-javascript-sdk-base/pull/81